azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -79,9 +79,19 @@ func resourceLinuxVirtualMachine() *schema.Resource {
 			"os_disk": virtualMachineOSDiskSchema(),
 
 			"size": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validation.StringIsNotEmpty,
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  validation.StringIsNotEmpty,
+				ConflictsWith: []string{"sizes"},
```

Comment:
`ConflictsWith` means these two attributes cannot co-exist, but it does not prevent these two could be both absent. Therefore could we change this by using `ExactlyOneOf`?
```suggestion
				ExactlyOneOf: []string{"size", "sizes"},
```
And the same for `sizes`